Library/Homebrew/command.rb

```diff
@@ -0,0 +1,134 @@
+module Homebrew
+  class Command
+    attr_reader :command_name, :valid_options, :description, :help_output, :man_output
+
+    def self.initialize
+      @valid_options = []
+      @description = nil
+      # TODO: set the @command_name dynamically
+      @command_name = "commands"
+      @help_output = nil
+      @man_output = nil
+    end
+
+    def self.options(&block)
+      initialize
+      @parent = nil
+      instance_eval(&block)
+      generate_help_and_manpage_output
+      puts "valid_options: ", @valid_options
+      # check_invalid_options(ARGV.options_only)
+    end
+
+    # def self.option(key, value = "No description for this option is available", **keyword_args)
+    #   children_options = keyword_args[:children_options]
+    #   @valid_options.push({option: key, desc: value, children_options: children_options})
+    # end
+
+    def self.add_valid_option(option, desc)
+      valid_option = {:option => option, :desc => desc, :children_options => nil}
```

Comment:
Another style nit: `child_options` is shorter and makes a bit more sense (options that are children, versus options *for* children).